src/KM_Controls.pas

```diff
@@ -2760,6 +2764,9 @@ function TKMEdit.KeyDown(Key: Word; Shift: TShiftState): Boolean;
   Result := KeyEventHandled(Key, Shift);
   if inherited KeyDown(Key, Shift) or ReadOnly then Exit;
 
+  //Allow some keys while blocking input
+  if BLockInput and not ((Key in [VK_LEFT, VK_RIGHT, VK_HOME, VK_END]) or ((ssCtrl in Shift) and (Key in [Ord('A'), Ord('C')]))) then Exit;

```

Comment:
Can you split this long condition into several lines ? It will be easier to read it.
Also pls fix this: 'BlockInput', instead of 'BLockInput'